kvmagent/ansible/kvm.py

```
@@ -113,6 +113,94 @@ def update_libvirtd_config(host_post_info):
+    libvirt_pki_dir = "%s/pki/libvirt" % zstack_lib_dir
+
+    # Step 1: Ensure local CA exists on the management node
+    command = (
+        "mkdir -p {ca_dir} {libvirt_pki_dir} && "
+        "if [ ! -f {ca_dir}/cacert.pem ]; then "
+        "  openssl genrsa -out {ca_dir}/cakey.pem 4096 && "
+        "  openssl req -new -x509 -days 3650 -key {ca_dir}/cakey.pem "
+        "    -out {ca_dir}/cacert.pem -subj '/O=ZStack/CN=ZStack Libvirt CA'; "
+        "fi"
+    ).format(ca_dir=ca_dir, libvirt_pki_dir=libvirt_pki_dir)
+    shell_return = os.system(command)
```

Ruff (0.15.5)

[error] Starting a process with a shell, possible injection detected

(S605)